### PR TITLE
ID select action should really filter on the IDs

### LINKED ISSUE #90: ID select action should really filter on the ID
the option `Select and filter object ID` actually creates a filter that makes a `match` query with the label. However, that query can match other documents.
Was this intended ? If yes, the action should be renamed. Otherwise, the created filter needs to be corrected.

### PR DESCRIPTION
**Pull request by giorgio94c**
Close #90

### CODE REVIEW

File: public/kibi_timeline.js [hunk 1]
```diff
@@ -261,6 +274,7 @@ define(function (require) {
                   }
 
                   const e =  {
+                    _id: hit._id,
```
Comment (scampi) [inline public/kibi_timeline.js]: take `hit._type` too in order to pass it to the `ids` query.

File: public/kibi_timeline_vis_params.html [hunk 2]
```diff
@@ -116,6 +116,9 @@
             <input type="radio" ng-model="vis.params.selectValue" value="id" /> Select and filter object ID
           </label>
           <label>
+            <input type="radio" ng-model="vis.params.selectValue" value="label" /> Select and filter by a label associated with object
```
Comment (scampi) [inline public/kibi_timeline_vis_params.html]: s/ a / the /
s/with object/with the event/

**Comment by giorgio94c** [comment]
Ok @scampi , I'm going to write tests for this...

File: public/kibi_timeline.js (hunk 1 above)
Comment (szydan) [inline public/kibi_timeline.js]: Done

File: public/webpackShims/vis.js [hunk 3]
```diff
@@ -1,2 +1,2 @@
 require('../../node_modules/vis/dist/vis.min.css');
-module.exports = require('../../node_modules/vis/dist/vis.min.js');
```
Comment (scampi) [inline public/webpackShims/vis.js]: why this change ?

File: public/__tests__/kibi_timeline.js [hunk 4]
```diff
@@ -138,6 +147,330 @@ describe('KibiTimeline Directive', function () {
...
+          values: ['61']
+        }
+      },
+      meta: {
+        index: 'logstash-*'
+      }
+    }];
+
+    // wait until timeline is fully rendered
+    setTimeout(function () {
```
Comment (scampi) [inline public/__tests__/kibi_timeline.js]: it would be better to avoid setTimeout since on some machine the timeline might not be rendered after 200ms.
can you use some `complete` event for that ?

File: public/__tests__/kibi_timeline.js [hunk 5]
```diff
@@ -138,6 +147,330 @@ describe('KibiTimeline Directive', function () {
...
+          // cheat that we clicked on the item
+          'timeline-item': {}
+        }
+      });
+      const eventData = {
+        target: $item[0],
+        srcEvent: event
+      };
+      // cheat again and trigger the tap with fake eventData
+      $panel[0].hammer[0].emit('tap', eventData);
```
Comment (scampi) [inline public/__tests__/kibi_timeline.js]: comment that hammer comes from the vis lib

File: public/kibi_timeline.js [hunk 6]
```diff
@@ -96,6 +96,20 @@ define(function (require) {
...
+            const q2 = {
+              query: {
+                ids: {}
+              },
+              meta: {
+                index: selected.index
+              }
+            };
+            q2.query.ids = {
+              type: selected._type,
```
Comment (scampi) [inline public/kibi_timeline.js]: not big problem, but why couldn't this be done on line 102 above ?

File: public/webpackShims/vis.js (hunk 3 above)
Comment (szydan) [inline public/webpackShims/vis.js]: To be honest it is a left over after a debugging session 
But I would leave it as in this way
1 it is easier to debug 
2 webpack will optimize anyway (if we leave the min I think you have to tell webpack to skip such a file during optimization)

**Comment by szydan** [comment]
@scampi I've tried the noDigest but it does not help, the only way to get it to work is to use this setTimeout 0
